ash/src/device.rs

```diff
@@ -328,6 +328,7 @@ impl Device {
             &mut count,
             out.as_mut_ptr(),
         );
+        assert_eq!(count, out.len() as u32);
```

Comment:
could use `as _` ?

Reply:
What's the advantage of not being explicit about this (trivial to write) type?

That said, it's probably better to cast `count` to `usize` to prevent a possible truncation.

Reply:
I am always wondering wether to use `as _` or be explicit...

Reply:
@filnet There are probably no real API guidelines but it's probably better to be explicit unless intentionally hiding long type casts.

I've also turned on the `trivial-casts` and `trivial-numereric-casts` default-`allow` `rustc` warning locally to notice that there are quite a few casts to the same type, mostly `as _` which was copied verbatim without the programmer really knowing or caring about the source and target type.

In either case we should aim to flip that warning on and make the code ever so slightly more readable by removing stray `as` casts.

> That said, it's probably better to cast `count` to `usize` to prevent a possible truncation.

On second thought I did this by intention to match the truncation from `let mut count = out.len() as u32;`. However, that means - if someone manages to pass a slice longer than `u32::max` at all - this assert won't fail even if it should. Erring on the side of correctness, this is now flipped.